x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx

```diff
@@ -146,13 +151,94 @@ const configurationFormSchema: FormSchema = {
       }
     ),
   },
+  dataRetentionEnabled: {
+    type: FIELD_TYPES.TOGGLE,
+    defaultValue: false,
+    label: i18n.translate(
+      'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.dataRetentionEnabledField',
+      {
+        defaultMessage: 'Enable data retention',
+      }
+    ),
+  },
 };
 
-export const EditDataRetentionModal: React.FunctionComponent<Props> = ({
-  lifecycle,
+interface MixedIndicesCalloutProps {
+  dslWithIlmIndices: {
+    ilmIndices: Array<{ name: string }>;
+    dslIndices: Array<{ name: string }>;
+  };
+  history: ScopedHistory;
+  ilmPolicyLink: string;
+  ilmPolicyName?: string;
+  dataStreamName: string;
+}
+
+const MixedIndicesCallout = ({
+  dslWithIlmIndices,
+  ilmPolicyLink,
+  ilmPolicyName,
   dataStreamName,
+  history,
+}: MixedIndicesCalloutProps) => {
+  const MAX_VISIBLE_INDICES = 3;
+  const visibleIndices = dslWithIlmIndices.ilmIndices.slice(0, MAX_VISIBLE_INDICES);
+
+  return (
+    <EuiCallOut
+      title={i18n.translate(
+        'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.someManagedByILMTitle',
+        { defaultMessage: 'Some indices are managed by ILM' }
+      )}
+      color="warning"
+      iconType="warning"
+      data-test-subj="someIndicesAreManagedByILMCallout"
+    >
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.someManagedByILMBody"
+          defaultMessage="The following indices are managed by ILM: {managedByILM} {hasManyIndices, plural, one {and {remainingIndices} more} other { } } ({viewAllIndicesLink}). Updating data retention will not affect these indices, instead you will have to update the {ilmPolicyLink} policy."
```

Comment:
```suggestion
          defaultMessage="The following indices are managed by ILM: {managedByILM} {hasManyIndices, plural, one {and {remainingIndices} more} other { } } [{viewAllIndicesLink}]. Updating data retention for this data stream won't affect these indices. Instead you will have to update the {ilmPolicyLink} policy."
```
Do we control labels of indices or can they be 100 characters long? In case it can be randomly long, maybe 2 alternatives to consider to have more control over readability for this callout:
- Don't display index names by default, and instead say: `One or several indices are managed by an ILM policy [view indices]. Updating data [...].`. I'm also thinking that this method might be better in case there would be multiple ILM policies managing different indices (not sure it's possible, but in case).
- Have a bullet list with the first 2 or 3 ones, and have a last bullet that contains the [view all indices] link. It'll make the callout longer, but easier to read.

Reply:
Thanks for the suggestion, I neded up going for the first option as the modal is already a big large to accomodate a bulleted list.

Reply:
👍 That was my favorite option too :)